rhoas/topics/datasource_topic.go

```diff
@@ -0,0 +1,77 @@
+package topics
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	rhoasAPI "redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/api"
+	"redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/utils"
+)
+
+func DataSourceTopic() *schema.Resource {
+	return &schema.Resource{
+		Description: "`rhoas_topic` provides a Topic accessible to your organization in Red Hat OpenShift Streams for Apache Kafka.",
+		ReadContext: dataSourceTopicRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the Kafka topic",
+			},
+			"partitions": {
+				Description: "The number of partitions in the topic",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"kafka_id": {
```

Comment:
I think we need to define end user use case we have in the roadmap.
We have method to create kafka and another method to create topic that require hardcoding kafka id. 
Not sure how creation of kafka and topics could be possible if we need to hardcode kafka id.

That is our happy day scenario (I want kafka with topics and permission to read from them

Reply:
Awesome. We should also collect all required connection details during every step generation:

- BootstrapURL + ID on Kafka Create
- ServiceAccountId + ServiceAccountClientId + ServiceAccountSecret
- Topic list

And let users have another steps that use that data to provision services that depend on kafka.  What it feels to me is that we going to have "example based workflow" where we would document user use cases as set of recipes instead of single step "create kafka and give me connection details" approach